.github/workflows/version.yml

```diff
@@ -0,0 +1,95 @@
+          sed -i "s/CURRENT = $CURRENT_VERSION_UNDERSCORE;/CURRENT = $NEXT_VERSION_UNDERSCORE;/g" server/src/main/java/org/opensearch/Version.java
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          base: ${{ env.BASE }}
+          branch: 'create-pull-request/patch-${{ env.BASE }}'
+          commit-message: Incremented version to ${{ env.NEXT_VERSION }}
+          delete-branch: true
+          title: '[AUTO] Incremented version to ${{ env.NEXT_VERSION }}.'
+          body: |
+            I've noticed that a new tag ${{ env.TAG }} was pushed, and incremented the version from ${{ env.CURRENT_VERSION }} to ${{ env.NEXT_VERSION }}.
+
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ env.BASE_X }}
+      - name: Add bwc version to .X branch
+        run: |
+          echo Adding bwc version $NEXT_VERSION after $CURRENT_VERSION
+          sed -i "s/- \"$CURRENT_VERSION\"/\0\n  - \"$NEXT_VERSION\"/g" .ci/bwcVersions
+          echo Adding $NEXT_VERSION_UNDERSCORE after $CURRENT_VERSION_UNDERSCORE
+          sed -i "s/public static final Version $CURRENT_VERSION_UNDERSCORE = new Version(\([[:digit:]]\+\)\(.*\));/\0\n    public static final Version $NEXT_VERSION_UNDERSCORE = new Version($NEXT_VERSION_ID\2);/g" server/src/main/java/org/opensearch/Version.java
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          base: ${{ env.BASE_X }}
+          branch: 'create-pull-request/patch-${{ env.BASE_X }}'
+          commit-message: Added bwc version ${{ env.NEXT_VERSION }}
+          delete-branch: true
+          title: '[AUTO] [${{ env.BASE_X }}] Added bwc version ${{ env.NEXT_VERSION }}.'
+          body: |
+            I've noticed that a new tag ${{ env.TAG }} was pushed, and added a bwc version ${{ env.NEXT_VERSION }}.
+
+      - uses: actions/checkout@v2
+        with:
+          ref: main
+      - name: Add bwc version to main branch
+        run: |
+          echo Adding bwc version $NEXT_VERSION after $CURRENT_VERSION
+          sed -i "s/- \"$CURRENT_VERSION\"/\0\n  - \"$NEXT_VERSION\"/g" .ci/bwcVersions
+          echo Adding $NEXT_VERSION_UNDERSCORE after $CURRENT_VERSION_UNDERSCORE
+          sed -i "s/public static final Version $CURRENT_VERSION_UNDERSCORE = new Version(\([[:digit:]]\+\)\(.*\));/\0\n    public static final Version $NEXT_VERSION_UNDERSCORE = new Version($NEXT_VERSION_ID\2);/g" server/src/main/java/org/opensearch/Version.java
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          base: main
+          branch: 'create-pull-request/patch-main'
+          commit-message: Added bwc version ${{ env.NEXT_VERSION }}
+          delete-branch: true
```

Comment:
@VachaShah you could use this for auto-backport workflow. 
This should solve the problem of stale branches after the PR is merged.

Reply:
Thanks @saratvemulapalli ! I just created a PR to fix the auto-backport workflow.